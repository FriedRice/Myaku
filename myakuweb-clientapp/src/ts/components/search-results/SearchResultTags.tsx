/** @module Search result tag list component */

import { ArticleSearchResult } from 'ts/types/types';
import React from 'react';

interface SearchResultTagsProps {
    searchResult: ArticleSearchResult;
}
type Props = SearchResultTagsProps;


const SearchResultTags: React.FC<Props> = function(props) {
    var tagLis: React.ReactElement[] = [];
    for (const tag of props.searchResult.tags) {
        tagLis.push(
            <li key={tag} className='tag-badge'>{tag}</li>
        );
    }

    return (
        <ul className='article-tags'>
            {tagLis}
        </ul>
    );
};

export default SearchResultTags;
