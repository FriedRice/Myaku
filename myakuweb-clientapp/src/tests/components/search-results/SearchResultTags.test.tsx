/**
 * Tests for the SearchResultTags component.
 * @module tests/components/search-results/SearchResultTags.test
 */

import { BASELINE_SEARCH_RESULT_PAGE } from 'tests/testData';
import React from 'react';
import SearchResultTags from 'ts/components/search-results/SearchResultTags';
import { shallow } from 'enzyme';


describe('<SearchResultTags />', function() {
    it('renders correctly', function() {
        const testResults = BASELINE_SEARCH_RESULT_PAGE.results;
        const wrapper = shallow(
            <SearchResultTags
                searchResult={testResults[0]}
            />
        );
        expect(wrapper).toMatchSnapshot();

        wrapper.setProps({searchResult: testResults[1]});
        expect(wrapper).toMatchSnapshot();
    });
});
