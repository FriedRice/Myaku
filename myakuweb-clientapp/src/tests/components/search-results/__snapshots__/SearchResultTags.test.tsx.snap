// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultTags /> renders correctly 1`] = `
<ul
  className="article-tags"
>
  <li
    className="tag-badge"
    key="Short length"
  >
    Short length
  </li>
  <li
    className="tag-badge"
    key="Cool"
  >
    Cool
  </li>
</ul>
`;

exports[`<SearchResultTags /> renders correctly 2`] = `
<ul
  className="article-tags"
>
  <li
    className="tag-badge"
    key="Medium length"
  >
    Medium length
  </li>
</ul>
`;
